hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/AuthenticationBaseMessageTask.java

```diff
@@ -134,6 +134,12 @@ public void processMessage() throws Throwable {
             return;
         }
         String uuid = getUuid();
+        if (clientMessage.getCorrelationId() < clientEngine.getLastAuthenticationCorrelationId(uuid)) {
```

Comment:
I was not hundred percent sure about this either. In the first version it was not implemented. I added after review comments. This is not requirement, just optimisation.
If correlation id is already updated, this prevents sending operations to cluster.  My thinking is that since authentication is not something common, doing optimisation here not that important. And more code always means more bug.